storage/range_test.go

```diff
@@ -134,7 +134,13 @@ func (tc *testContext) Start(t *testing.T) {
 	tc.stopper.AddCloser(tc.transport)
 
 	if tc.store == nil {
-		tc.store = NewStore(tc.clock, tc.engine, nil, tc.gossip, tc.transport, TestStoreConfig)
+		ctx := TestStoreContext
+		ctx.Clock = tc.clock
+		ctx.Gossip = tc.gossip
+		ctx.Transport = tc.transport
+		// Dummy DB.
+		ctx.DB = &client.KV{}
```

Comment:
I'd rather remove the nil check for ctx.DB in StoreContext.Valid than pass in a zero-valued `client.KV` here. We've been fine allowing nil for the `db` parameter so far and that's better than passing a value that looks valid but won't actually work.